types/shopify-buy/tslint.json

```diff
@@ -1,24 +1,6 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
-        "array-type": false,
-        "ban-types": false,
-        "no-consecutive-blank-lines": false,
-        "no-declare-current-package": false,
-        "no-padding": false,
-        "no-single-declare-module": false,
-        "no-unnecessary-qualifier": false,
-        "no-var-keyword": false,
-        "object-literal-shorthand": false,
-        "one-variable-per-declaration": false,
-        "only-arrow-functions": false,
-        "prefer-const": false,
-        "prefer-template": false,
-        "semicolon": false,
-        "space-before-function-paren": false,
-        "space-within-parens": false,
-        "strict-export-declare-modifiers": false,
-        "typedef-whitespace": false,
-        "whitespace": false
+        "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
```

Comment:
Why do you need to turn off this rule?

Reply:
typescript expected `export =` the end of file. I am not sure, because the library uses `export default`. Do you have any idea how to remove this rule? I would like to remove that but I am not sure how.

Reply:
I looked at the library, and what happens is that, in `package.json`, `main` points to `index.js`, and `index.js` uses `module.exports = ...`, therefore `export =` should be used in the .d.ts file. However, the same `package.json` also has a `jsnext:main` field pointing to `index.es.js`, which indeed uses `export default`. Now, since `jsnext:main` is a field in `package.json` that seems to be supported by specific bundlers, we don't have a way of supporting that. If the original package used export maps to provide both commonjs and esm entry points, then we could do the same here...
In light of that, I think you should use `export =` in the .d.ts file, but I'm open to hearing your opinion on that.

Reply:
I would like to export all types as well. Not just the one method. Is there any way how to do it?

Reply:
That depends. How are the types going to be imported/used? The way the typings were written before is one of the ways, where you have something like:

```ts
declare namespace Foo {
    type Baz = { b: "b" };
}
declare function Foo(a: number, b: number): number;
export = Foo;
```

and then when you do `import Foo = require("foo")`, you can access types like so: `Foo.Baz`.
But maybe your question is about how to do re-exports of the types that are now being imported into the main `index.d.ts`?

Reply:
thanks for your review and help but I am still not able to export types. Please could you write an example of how should it be right?

Reply:
I have tried thinking of ways of doing it, but I couldn't really think of anything besides manually importing the types from the other files and exporting them from inside the `ShopifyBuy` namespace in `index.d.ts`. Maybe it's just too complicated to refactor this package in the way you intended to? 😕

Reply:
@gabritto Thanks a lot for your help. Could you please review it?